docs/labels.md

```diff
@@ -11,50 +10,66 @@ key/value labels set on it, with at most one label with a particular key.
+The API currently supports two types of selectors: equality based and set based.
+A label selector can be made of multiple _requirements_ which are comma separated. In case of multiple requirements, all must be satisfied so comma separaator acts as an AND logical operator.
+
+
+### Equality based requirement
+
+Equality (or inequality) based requirements allow filtering by label keys and values. Matching objects must have all of the specified labels (both keys and values), though they may have additional labels as well.
+Three kinds of operators are admitted `=`,`==`,`!=`. The first two represent _equality_ and are simply synonyms. While the latter represents _inequality_. For example:
 ```
-"selector": {
-  "environment": "dev",
-  "tier": "frontend"
-}
+environment = production
+tier != frontend
 ```
 
-We [are actively working](https://github.com/GoogleCloudPlatform/kubernetes/issues/341) to support more types of selectors in the API, such as:
+The former selects all resources with key equal to `environment` and value equal to `production`.
+The latter selects all resources with key equal to `tier` and value distinct from `frontend`.
+One could filter for resources in `production` but not `frontend` using the comma operator: `environment=production,tier!=frontend`
+
+### Set based requirement
+
+Set based label requirements allow filtering keys according to a set of values. Matching objects must have all of the specified labels (i.e. all keys and at least one of the values specified for each key). Three kind of operators are supported: `in`,`notin` and exists (only the key identifier). For example:
 ```
-key1
-key1 = value11
-key1 != value11
-key1 in (value11, value12, ...)
-key1 not in (value11, value12, ...)
+environment in (production, qa)
+tier notin (frontend, backend)
+partition
 ```
+The first example selects all resources with key equal to `environment` and value equal to `production` or `qa`.
+The second example selects all resources with key equal to `tier` and value other than `frontend` and `backend`.
+The third example selects all resources including a label with key `partition`; no values are checked.
+Similary the comma separator acts as an _AND_ operator for example filtering resource with a `partition` key (not matter the value) and with `environment` different than  `qa`. For example: `partition,environment notin (qa)`.
+The set based label selector is a general form of equality since `environment=production` is equivalent to `environment in (production)`; similarly for `!=` and `notin`.
+
+Set based requirements can be mixed with equality based requirements. For example: `partition,environment!=qa`.
+
 
 LIST and WATCH operations may specify label selectors to filter the sets of objects returned using a query parameter: `?labels=key1%3Dvalue1,key2%3Dvalue2,...`.
 
-The `service` and `replicationController` kinds of objects use selectors to match sets of pods that they operate on.
```

Comment:
Somewhere here we need a short section on selectors in API objects, such as service and replication controller, with an example using the map syntax and clarifying that only equality-based selectors are currently supported.